### PR TITLE
Cherry-pick PR #34909

### PR DESCRIPTION
**Pull request by RikkiGibson**
Based on #34909. Meant to exercise #35217.

cc @gafter @jaredpar

We believe we've addressed the problem that this change produces via #35311 and #35373.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs [hunk 1]
```diff
@@ -20,65 +20,52 @@ internal static class NullableAnnotationExtensions
         /// Join nullable annotations from the set of lower bounds for fixing a type parameter.
         /// This uses the covariant merging rules.
         /// </summary>
-        public static NullableAnnotation Join(this NullableAnnotation a, NullableAnnotation b)
-        {
-            if (a.IsAnnotated() || b.IsAnnotated())
-                return NullableAnnotation.Annotated;
-            return (a < b) ? a : b;
-        }
+        public static NullableAnnotation Join(this NullableAnnotation a, NullableAnnotation b) => (a < b) ? b : a;
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Symbols/NullableAnnotationExtensions.cs]: >Join [](start = 41, length = 4)

Could you update the doc comment to say something like "Annotated wins over Oblivious which wins over NotAnnotated"?
Unless people memorize the enum order, the formula does not really help understand the behavior.
(Same for `Meet` below, if you don't mind)

**Comment by RikkiGibson** [comment]
Planning on merging this after the dev16.2 preview1 snap on May 8th

**Comment by jcouv** [comment]
@RikkiGibson This PR seems ready to merge, if it is still relevant. Can it be merged or closed?

**Comment by RikkiGibson** [comment]
It's definitely relevant. I was just waiting until things calmed down with VS insertions because there's a small risk that this re-introduces a build failure on the VS side. Since we just inserted to all VS branches (yay!) I will go ahead and merge now.

**Pull request merged**